tools/cache.py

```diff
@@ -46,14 +46,17 @@ def get(self, shortname, creator, extension='.bc', what=None):
 
 # Given a set of functions of form (ident, text), and a preferred chunk size,
 # generates a set of chunks for parallel processing and caching.
-def chunkify(funcs, chunk_size, DEBUG=False):
+def chunkify(funcs, chunk_size, DEBUG=False, using_fn_implementations=False):
```

Comment:
The reason for this is that the DFE algorithm passes in function bodies (e.g. function a { ... }), while the other passes seem to be passing in function identifiers followed by the function bodies as an array. I think this is how I was originally doing it as well, but I recall finding a small bug in split_funcs - in some rare cases, it seems to not do the right thing because of this line:

    parts = map(lambda part: part, js.split('\n}\n'))

This matching fails for a very small subset of generated functions in some rare cases where there may be a space between the \n and the }. Unfortunately I cannot repro this at the moment however. As a result, I ended up writing my own function splitter (see code block assigning funcs variable in duplicate_function_eliminator.py). 

All said and done though, I think it should be straightforward to revert the changes to cache.py and rewrite the code block that I wrote to generate the functions to also do the same thing that split_funcs was doing. Even better would be to identify the bug :-). I'll see if I can get a repro again.